include/rmm/mr/device/detail/arena.hpp

```diff
@@ -279,13 +275,16 @@ class global_arena final {
     RMM_EXPECTS(maximum_size_ == default_maximum_size || maximum_size_ == align_up(maximum_size_),
                 "Error, Maximum arena size required to be a multiple of 256 bytes");
 
-    std::size_t free{}, total{};
-    RMM_CUDA_TRY(cudaMemGetInfo(&free, &total));
-    if (initial_size == default_initial_size) {
-      initial_size = align_up(std::min(free, total / 2));
+    if (initial_size == default_initial_size || maximum_size == default_maximum_size) {
+      std::size_t free{}, total{};
+      RMM_CUDA_TRY(cudaMemGetInfo(&free, &total));
+      if (initial_size == default_initial_size) {
```

Comment:
Consider using `thrust::optional` as in #601, and also use upstream->get_mem_info() for MRs that support it.

Reply:
Seems like that would break existing callers? Mayer in a followup PR once #601 is merged?

Reply:
Yeah that's why I approved.